compiler/front/optionsprocessor.nim

```diff
@@ -0,0 +1,1698 @@
+## this modules process various options given as a flags and applies them to
+## a given `ConfigRef`. This allows for command line, configuration, comptime,
+## etc configuration building and updates through a unified interface
```

Comment:
```suggestion
## this module processes various options given as flags and applies them to
## a given `ConfigRef`. This allows for command line, configuration, compile-time,
## etc configuration building and updates through a unified interface
```
'comptime' -> 'compile-time' is only a soft suggestion -- I think it makes sense to write it out in documentation.

Reply:
I'm aiming to make `comptime` part of the regular vernacular.